File: src/api/launch.ts
import { invoke } from "@tauri-apps/api/core";

export const openInEditor = (worktreePath: string, branch: string) =>
  invoke<void>("open_in_editor", { worktreePath, branch });

export const openInTerminal = (worktreePath: string, branch: string) =>
  invoke<void>("open_in_terminal", { worktreePath, branch });

File: src-tauri/src/api/launch.rs
use crate::domain::error::Result;
use crate::service;

#[tauri::command]
pub async fn open_in_editor(worktree_path: String, branch: String) -> Result<()> {
    super::blocking(move || service::launch::open_in_editor(&worktree_path, &branch)).await
}

#[tauri::command]
pub async fn open_in_terminal(worktree_path: String, branch: String) -> Result<()> {
    super::blocking(move || service::launch::open_in_terminal(&worktree_path, &branch)).await
}

#[tauri::command]
pub async fn open_in_file_manager(worktree_path: String) -> Result<()> {
    super::blocking(move || service::launch::open_in_file_manager(&worktree_path)).await
}
